### ISSUE TITLE
userData not working in IE7

### ISSUE THREAD

**Issue by HitmanInWis** [medialize/jQuery-store#3]
Is userData working for anyone in IE7?  When I test, IE7 attempts to use the userData driver, but fails on init().  The line of code that's failing is:

  document.documentElement.insertBefore( this.element, document.getElementsByTagName('title')[0] );

If I change that line to:

  document.documentElement.insertBefore( this.element, document.getElementsByTagName('body')[0] );

the init() function no longer fails, but storage isnt working.

Ideas?

**Comment by rodneyrehm** [comment]
Make sure to initialize the store onDOMReady (or onLoad) e.g. with $.ready() or make sure your <script>s are at the end of <body>. The userData behavior relies on the DOM, so you can't use it before it's ready.

**Comment by HitmanInWis** [comment]
I am indeed loading storage within a call to $(document).ready().  To make sure it wasn't anything in my code causing the issue, I created a simple HTML page where I imported jquery-1.6.1 and the storage plugin.

  $(document).ready(function() {
      $.storage = new $.store();
      alert($.storage.driver.ident);
  });

In IE7 this is returning $.store.drivers.windowName

Could it be my version of jquery?

**Comment by rodneyrehm** [comment]
http://jsfiddle.net/wm48g/2/ runs fine in my IE7.

**Comment by HitmanInWis** [comment]
Ah...thought I had verified I had the latest version of jquery.store.js, but it looks like you have a newer version on there that is indeed working for my IE7.  I'll have to copy the new version down - thanks!

**Comment by HitmanInWis** [comment]
Rodney,

I was experiencing an issue with the userData flush function.  If a script called flush, and then immediately after started adding data to storage, those new values would be considered expired.  When I tried to retrieve the values from storage, they weren't there.

To fix the problem, I modified the flush() function for userData.

```
    flush: function()
    {
        var attrs = this.element.xmlDocument.firstChild.attributes;
        for (var i = attrs.length - 1; i >= 0; i--) {
            this.element.removeAttribute( attrs[i].nodeName );
        }
        this.element.save( this.nodeName );
    }
```

Feel free to incorporate the change into your codebase if you think it's a good idea.
